syntax = "proto3";

// Import wrappers for simple types like string and int
import "google/protobuf/wrappers.proto";

package pb;

option go_package = "./pb"; // This is for Go, but it's good practice

service AccountService {
  rpc Register (RegisterRequest) returns (google.protobuf.StringValue);
  rpc Login (LoginRequest) returns (google.protobuf.StringValue);
  rpc GetAccount (google.protobuf.UInt64Value) returns (AccountResponse);
}

// --- Messages ---

message Account {
  uint64 id = 1;
  string name = 2;
  string email = 3;
}

message RegisterRequest {
  string name = 1;
  string email = 2;
  string password = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message AccountResponse {
  Account account = 1;
}